modules/edit-modes/src/lib/measure-distance-mode.ts

```diff
@@ -1,137 +1,180 @@
-// @ts-ignore
 import turfDistance from '@turf/distance';
-// @ts-ignore
+import { FeatureCollection } from '../geojson-types';
+import { ClickEvent, PointerMoveEvent, ModeProps, GuideFeatureCollection, Tooltip } from '../types';
+import { getPickedEditHandle } from '../utils';
 import memoize from '../memoize';
-import {
-  ClickEvent,
-  PointerMoveEvent,
-  Tooltip,
-  ModeProps,
-  GuideFeatureCollection,
-  EditHandleFeature,
-} from '../types';
-import { FeatureCollection, Position } from '../geojson-types';
 import { GeoJsonEditMode } from './geojson-edit-mode';
 
-const DEFAULT_TOOLTIPS = [];
-
 export class MeasureDistanceMode extends GeoJsonEditMode {
-  startingPoint: Readonly<EditHandleFeature> | null | undefined = null;
-  endingPoint: Readonly<EditHandleFeature> | null | undefined = null;
-  endingPointLocked = false;
+  _isMeasuringSessionFinished = false;
+  _currentTooltips = [];
+  _currentDistance = 0;
 
-  _setEndingPoint(mapCoords: Position) {
-    this.endingPoint = {
-      type: 'Feature',
-      properties: {
-        guideType: 'editHandle',
-        editHandleType: 'existing',
-        featureIndex: -1,
-        positionIndexes: [],
-      },
-      geometry: {
-        type: 'Point',
-        coordinates: mapCoords,
-      },
-    };
-  }
+  _calculateDistanceForTooltip = memoize(({ positionA, positionB, modeConfig }) => {
+    const { turfOptions, measurementCallback } = modeConfig || {};
+    const distance = turfDistance(positionA, positionB, turfOptions);
 
-  _getTooltips = memoize(({ modeConfig, startingPoint, endingPoint }) => {
-    let tooltips = DEFAULT_TOOLTIPS;
+    if (measurementCallback) {
+      measurementCallback(distance);
+    }
 
-    if (startingPoint && endingPoint) {
-      const { formatTooltip, turfOptions, measurementCallback } = modeConfig || {};
-      const units = (turfOptions && turfOptions.units) || 'kilometers';
+    return distance;
+  });
 
-      const distance = turfDistance(startingPoint, endingPoint, turfOptions);
+  _formatTooltip(distance, modeConfig?) {
+    const { formatTooltip, turfOptions } = modeConfig || {};
```

Comment:
Nit: Is `turfOptions` a new option? Generally we should document options, and write tests for them. 

Do we really want to expose the fact that we use `turf` underneath (seems like an implementation detail)?

Alternatively, can we just always return meters and degrees and app can make any other conversions?

Reply:
`turfOptions` was already in place (just don't run git blame on it). =)

I agree we should abstract that we use turf under the covers, but I'm OK with that being done in a separate PR.